File: src/org/webfabric/sitemesh/PathInfo.java
package org.webfabric.sitemesh;

import javax.servlet.http.HttpServletRequest;

public class PathInfo implements Path {
    private final String value;

    public PathInfo(HttpServletRequest request) {
        this.value = request.getPathInfo();
    }

    public PathInfo(String value) {
        this.value = value;
    }

    public String value() {
        return value;
    }
}

File: src/org/webfabric/servlet/PathInfo.scala
package org.webfabric.servlet;

import org.webfabric.io.{HierarchicalPath}
import javax.servlet.http.HttpServletRequest;

class PathInfo(v: String) extends HierarchicalPath(v)

object PathInfo {
  def apply(request: HttpServletRequest):PathInfo = {
    return new PathInfo(request.getPathInfo());
  }
}
